### PR TITLE
Various profiles

### PR DESCRIPTION
**Pull request by kortewegdevries**

### CODE REVIEW

File: etc/inc/disable-programs.inc [hunk 1]
```diff
@@ -426,6 +428,7 @@ blacklist ${HOME}/.frozen-bubble
 blacklist ${HOME}/.gimp*
 blacklist ${HOME}/.gist
 blacklist ${HOME}/.gitconfig
+blacklist ${HOME}/.gnome/apps
```
Comment (rusty-snake) [inline etc/inc/disable-programs.inc:431]: Breaks. disable-common.inc has `read-only ${HOME}/.gnome/apps`. Remove.

File: etc/profile-a-l/kazam.profile [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Firejail profile for kazam
+# Description: Screen capture tool
+# This file is overwritten after every install/update
+# Persistent local customizations
+include kazam.local
+# Persistent global definitions
+include globals.local
+
+ignore noexec ${HOME}
```
Comment (rusty-snake) [inline etc/profile-a-l/kazam.profile:9]: Does it need exec HOME?

File: etc/profile-a-l/kazam.profile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc 
+include disable-interpreters.inc
+include disable-programs.inc
+include disable-passwdmgr.inc
+include disable-shell.inc
+include disable-xdg.inc
+
+include whitelist-runuser-common.inc
```
Comment (rusty-snake) [inline etc/profile-a-l/kazam.profile:27]: wusc?

File: etc/profile-m-z/menulibre.profile [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Firejail profile for menulibre
+# Description: Create desktop and menu launchers easily
+# This file is overwritten after every install/update
+# Persistent local customizations
+include menulibre.local
+# Persistent global definitions
+include globals.local
+
+ignore read-only ${HOME}/.local/share/applications
+ignore read-only ${HOME}/.config/menus
```
Comment (rusty-snake) [inline etc/profile-m-z/menulibre.profile:10]: Replace with `read-write`.

File: etc/profile-m-z/menulibre.profile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+noblacklist ${HOME}/.gnome/apps
```
Comment (rusty-snake) [inline etc/profile-m-z/menulibre.profile:12]: remove noblacklist and add read-write

File: etc/profile-m-z/menulibre.profile [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc 
+include disable-interpreters.inc
+include disable-programs.inc
+include disable-passwdmgr.inc
+include disable-xdg.inc
+
+include whitelist-runuser-common.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/menulibre.profile:25]: wusc?

whitelist?

File: etc/profile-a-l/kazam.profile [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+include disable-xdg.inc
+
+include whitelist-runuser-common.inc 
+include whitelist-var-common.inc
+
+apparmor
+caps.drop all
+net none
+nodvd
+no3d
```
Comment (rusty-snake) [inline etc/profile-a-l/kazam.profile:34]: Do it need hw-accel?

File: etc/profile-a-l/kazam.profile [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+nonewprivs
+nogroups
+noroot
```
Comment (rusty-snake) [inline etc/profile-a-l/kazam.profile:37]: Sort

File: etc/profile-m-z/onboard.profile [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+apparmor
+caps.drop all
+machine-id
+net none
+nodvd
+no3d
+nonewprivs
+nogroups
+noroot
```
Comment (rusty-snake) [inline etc/profile-m-z/onboard.profile:39]: sort

File: etc/profile-a-l/kazam.profile (hunk 2 above)
Comment (kortewegdevries) [reply etc/profile-a-l/kazam.profile:9]: Yeah,python modules I think.

File: etc/inc/disable-programs.inc (hunk 1 above)
Comment (kortewegdevries) [reply etc/inc/disable-programs.inc:431]: Thanks,

File: etc/profile-m-z/menulibre.profile (hunk 5 above)
Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:12]: Thanks,

Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:12]: Okay.

File: etc/profile-m-z/menulibre.profile (hunk 4 above)
Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:10]: Okay.

File: etc/profile-a-l/kazam.profile (hunk 7 above)
Comment (kortewegdevries) [reply etc/profile-a-l/kazam.profile:34]: Recording screen needs HW? Also do you need to be part of audio group to record audio? Then `nogroups` needs to be removed.

File: etc/profile-a-l/kazam.profile (hunk 3 above)
Comment (kortewegdevries) [reply etc/profile-a-l/kazam.profile:27]: Thanks, wlll add.

File: etc/profile-m-z/menulibre.profile (hunk 6 above)
Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:25]: Half of the list is not being populated with desktop entries, tried adding `whitelist /usr/share/applications`
`whitelist /usr/share/menulibre` with wusc.

Comment (rusty-snake) [reply etc/profile-m-z/menulibre.profile:25]: Works the following? If not try with `--build`. If thats still don't work, leave it.
```
whitelist /usr/share/applications
whitelist /usr/share/menulibre
whitelist /usr/share/desktop-directories
include wusc.inc
```

File: etc/profile-a-l/kazam.profile (hunk 7 above)
Comment (rusty-snake) [reply etc/profile-a-l/kazam.profile:34]: > Recording screen needs HW?

If the record is encode as e.g. mp4.

> Also do you need to be part of audio group to record audio? Then nogroups needs to be removed.

This is usually done with pulse.

File: etc/profile-m-z/menulibre.profile (hunk 6 above)
Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:25]: Works,thanks but some of the entries have no icon picture. Tried adding `~./local/share/icons,/usr/share/icons,pixmaps,mime` etc. Build doesn't give anything more than this.

File: etc/profile-a-l/kazam.profile (hunk 7 above)
Comment (kortewegdevries) [reply etc/profile-a-l/kazam.profile:34]: I'll remove `no3d`.

File: etc/profile-m-z/menulibre.profile (hunk 6 above)
Comment (rusty-snake) [reply etc/profile-m-z/menulibre.profile:25]: flatpak: `/var/lib/flatpak/exports/share/{applications,icons}` and `~/.local/share/flatpak/exports/share/{applications,icons}`

> some of the entries have no icon picture.

Are these icons specified by name or by path?

Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:25]: Some them name; by custom path will be broken for some. Thanks for snap/flatpak.

Comment (rusty-snake) [reply etc/profile-m-z/menulibre.profile:25]: Can you `locate` for the name?

Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:25]: Found them.. Most of them are in, `/var/lib/app-info/icons/. Does this vary by distro?

Comment (rusty-snake) [reply etc/profile-m-z/menulibre.profile:25]: Fedora seems to use `/usr/share/app-info/`.

Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:31]: Arch uses `/usr/share/icons/` in `hicolor`

Comment (rusty-snake) [reply etc/profile-m-z/menulibre.profile:31]: FWIW:
- Snap: don't use bubblewrap
- flatpak: relies on bubblewrap
- nautilus: runs thumbnailers in a sandbox with a read-only filesystem

Comment (kortewegdevries) [reply etc/profile-m-z/menulibre.profile:31]: > flatpak: relies on bubblewrap

Yeah but not atril, liferea,software-properties-common/-gtk,update-manager. etc. These are from Ubuntu's own repo and shouldn't require a sandbox if there's no flatpak. [Idk what they are doing.](https://www.bleepingcomputer.com/news/security/ubuntu-is-undoing-a-gnome-security-feature/)

**Comment by rusty-snake** [comment]
I lost the overview, anything left?

**Comment by kortewegdevries** [comment]
> I lost the overview, anything left?

I forgot, I thought you kept one unresolved for the icon dirs.

**Comment by kortewegdevries** [comment]
According to snap docs desktop files are copied to usual dirs: https://snapcraft.io/docs/desktop-menu-icon-support, so I think it's good to go. If a snap author chooses to override this, a custom path can also be set which is outside our scope. I've removed the unrelated comments...

**Comment by kortewegdevries** [comment]
<...>
>> I lost the overview, anything left?

>I forgot, I thought you kept one unresolved for the icon dirs.

According to snap docs desktop files are copied to usual dirs: https://snapcraft.io/docs/desktop-menu-icon-support, so I think it's good to go. If a snap author chooses to override this, a custom path can also be set which is outside our scope. I've removed the unrelated comments...

**Comment by rusty-snake** [comment]
Thanks for this contribution!

**Pull request merged**